Homework.py

```diff
@@ -1,107 +0,0 @@
-#Task 1
-class Person:
-  def __init__(self, first_name, lastname, age):
-    self.firstname = first_name
-    self.lastname = lastname
-    self.age = age
-    self.assigned = False #may cause trouble if not attached to children inits
-  def __repr__(self):
-    return (self.firstname + " " + self.lastname )
-  def type_int(variable): #ask if I should create something to verify that the user input an integer  or if I should just trust that it will be ok for simplicity sake
-    return
-  def Add():
-    typep = input("Would you like to add a student or a teacher? \n Please type 'student' or 'teacher': ")
-    if typep == "student":
-      print("Great! What is their ...")
-      first = input('First Name: ')
-      last = input('Last Name: ')
-      age = input('Age: ')
-      year = input('Year: ')
-      person = Student(year, first, last, age)
-      print("Thank you:)")
-      roster = person.students
-      #roster = [a for a in dir(person) if not a.startswith('__') if type(a) == list] #let's assume this grabs the class/teacher roster, I want to see if it is possible to access the list of students/teachers in studen in this way
-      roster.append(person)
-      # print(roster)
-      # print(person.__class__.__name__)
-    elif typep == "teacher":
-      print("Great! What is their ...")
-      first = input('First Name: ')
-      last = input('Last Name: ')
-      age = input('Age: ')
-      salary = input('Salary: ')
-      subject = input('Subject: ')
-      person = Teacher(salary, subject, first, last, age)
-      print("Thank you:)")
-      roster = person.teachers
-      #roster = [a for a in dir(person) if not a.startswith('__')] #let's assume this grabs the class/teacher roster
-      roster.append(person)
-    else:
-      print("I'm sorry that is not an option")
-class Student(Person):
-  _students = []
-  def __init__(self, year, *args, **kwargs):
-    super().__init__(*args, **kwargs)
-    self.year = year
-  # def Add_Student(self):
-  #   Student.students.append(self)
-class Teacher(Person):
-  _teachers = []
-  def __init__(self, salary, subject,*args, **kwargs):
-    super().__init__(*args, **kwargs)
-    self.salary = salary
-    self.subject = subject
-  # def Add_Teacher(self):
-  #   Teacher.teachers.append(self)
-class Classroom(Student, Teacher):
```

Comment:
When thinking about inheritance, ask a question like: "is student a person?" If the answer is "yes" -> Student should inherit from Person. In this case, Classroom is neither a Student nor Teacher, therefore, it should not inherit from (Student, Teacher). What you use here is composition, not inheritance.